packages/trader/src/Modules/Trading/Containers/__tests__/trade.spec.tsx

```diff
@@ -0,0 +1,168 @@
+    });
+
+    it('should show market is closed overlay', () => {
+        rootStore.modules.trade.is_market_closed = true;
+        rootStore.modules.trade.has_only_forward_starting_contracts = false;
+        render(<MockTrader />);
+        expect(screen.getByText('This market is closed')).toBeInTheDocument();
+    });
+
+    it('should call unMount and onUnmount when component is mounted and unmounted', () => {
+        const onMount = jest.fn();
+        const onUnmount = jest.fn();
+        rootStore.modules.trade.onMount = onMount;
+        rootStore.modules.trade.onUnmount = onUnmount;
+        const { unmount } = render(<MockTrader />);
+        expect(onMount).toBeCalledTimes(1);
+        unmount();
+        expect(onUnmount).toBeCalledTimes(1);
+    });
+
+    it('should render chart loader when chart is loading', () => {
+        rootStore.modules.trade.is_chart_loading = true;
+        render(<MockTrader />);
+        expect(screen.getByText('ChartLoader')).toBeInTheDocument();
+    });
+
+    it('should not render chart loader when chart is not loading', () => {
+        rootStore.modules.trade.is_chart_loading = false;
+        render(<MockTrader />);
+        expect(screen.queryByText('ChartLoader')).not.toBeInTheDocument();
+    });
+
+    it('should have a correct height offset for div100vhcontainer for accumulator', () => {
+        rootStore.modules.trade.is_accumulator = true;
+        (isDesktop as jest.Mock).mockReturnValue(false);
+        render(<MockTrader />);
+        expect(screen.getByTestId('dt_div_100_vh')).toHaveStyle('height: calc(768px - 295px);');
+    });
+
+    it('should have a correct height offset for div100vhcontainer for turbos', () => {
+        rootStore.modules.trade.is_accumulator = false;
+        rootStore.modules.trade.is_turbos = true;
+        (isDesktop as jest.Mock).mockReturnValue(false);
+        render(<MockTrader />);
+        expect(screen.getByTestId('dt_div_100_vh')).toHaveStyle('height: calc(768px - 300px);');
+    });
+
+    it('should set category and subcategory when component is mounted', async () => {
+        rootStore.modules.trade.getFirstOpenMarket = jest.fn(() => ({
+            category: 'category',
+            subcategory: 'subcategory',
+        }));
+        await act(async () => {
+            render(<MockTrader />);
+        });
+        expect(rootStore.modules.trade.getFirstOpenMarket).toBeCalledTimes(1);
+    });
+
+    it('should be able to click see open markets button in market is closed overlay', async () => {
+        rootStore.modules.trade.getFirstOpenMarket = jest.fn(() => ({
+            category: 'category',
+            subcategory: 'subcategory',
+        }));
+        await act(async () => {
+            render(<MockTrader />);
+
+            const seeOpenMarketsButton = screen.getByText('See open markets');
+            expect(seeOpenMarketsButton).toBeInTheDocument();
+            seeOpenMarketsButton.click();
```

Comment:
Usually, after click you expect something)
Or checking that button was absent/disabled and after some steps it was enabled